### PR TITLE
feat(security): add unit tests for client_negotiation

### PR DESCRIPTION
**Pull request by levy5307**
1.add unit tests for `client_negotiation`
2.modify some unit tests for `server_negotiation`

### CODE REVIEW

File: src/runtime/test/client_negotiation_test.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+TEST_F(client_negotiation_test, on_recv_mechanisms)
+{
+    struct
+    {
+        negotiation_status::type resp_status;
+        std::string resp_msg;
+        std::string selected_mechanism;
+    } tests[] = {{negotiation_status::type::SASL_SELECT_MECHANISMS, "GSSAPI", ""},
+                 {negotiation_status::type::SASL_LIST_MECHANISMS_RESP, "TEST", ""},
+                 {negotiation_status::type::SASL_LIST_MECHANISMS_RESP, "GSSAPI", "GSSAPI"}};
```
Comment (acelyc111) [inline src/runtime/test/client_negotiation_test.cpp:67]: Add test case to test response with multiple mechanisms

Comment (levy5307) [reply src/runtime/test/client_negotiation_test.cpp:67]: good idea.

File: src/runtime/test/server_negotiation_test.cpp [hunk 2]
```diff
@@ -47,6 +48,9 @@ class server_negotiation_test : public testing::Test
 
     void on_select_mechanism(negotiation_rpc rpc) { _srv_negotiation->on_select_mechanism(rpc); }
 
+    // _sim_session is used for holding the sim_rpc_session which is created in ctor,
+    // in case it is released. Because negotiation keeps only a raw pointer.
+    rpc_session_ptr _sim_session;
     server_negotiation *_srv_negotiation;
```
Comment (neverchanje) [inline src/runtime/test/server_negotiation_test.cpp:54]:
```suggestion
    std::unique_ptr<server_negotiation> _srv_negotiation;
```

File: src/runtime/test/client_negotiation_test.cpp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    const std::string &get_selected_mechanism() { return _client_negotiation->_selected_mechanism; }
+
+    negotiation_status::type get_negotiation_status() { return _client_negotiation->_status; }
+
+    // _sim_session is used for holding the sim_rpc_session which is created in ctor,
+    // in case it is released. Because negotiation keeps only a raw pointer.
+    rpc_session_ptr _sim_session;
+    client_negotiation *_client_negotiation;
```
Comment (neverchanje) [inline src/runtime/test/client_negotiation_test.cpp:55]:
```suggestion
    std::unique_ptr<client_negotiation> _client_negotiation;
```

**Review by acelyc111 — DISMISSED** [review]

**Review by acelyc111 — APPROVED** [review]

**Review by neverchanje — APPROVED** [review]

**Pull request merged**